tests/components/konnected/test_config_flow.py

```diff
@@ -0,0 +1,1060 @@
+            "2": "Binary Sensor",
+            "3": "Digital Sensor",
+            "4": "Switchable Output",
+            "6": "Binary Sensor",
+            "7": "Digital Sensor",
+            "8": "Switchable Output",
+            "alarm1": "Switchable Output",
+            "out1": "Switchable Output",
+        },
+        "blink": True,
+        "binary_sensors": [
+            {"zone": "2", "type": "door", "inverse": False},
+            {"zone": "6", "type": "window", "name": "winder", "inverse": True},
+            {"zone": "10", "type": "door", "inverse": False},
+        ],
+        "sensors": [
+            {"zone": "3", "type": "dht", "poll_interval": 3},
+            {"zone": "7", "type": "ds18b20", "name": "temper", "poll_interval": 3},
+            {"zone": "11", "type": "dht", "poll_interval": 3},
+        ],
+        "switches": [
+            {"activation": "high", "zone": "4"},
+            {
+                "zone": "8",
+                "name": "switcher",
+                "activation": "low",
+                "momentary": 50,
+                "pause": 100,
+                "repeat": 4,
+            },
+            {"activation": "high", "zone": "out1"},
+            {"activation": "high", "zone": "alarm1"},
+        ],
+    }
+
+
+async def test_option_flow_import(hass, mock_panel):
+    """Test config flow options imported from configuration.yaml."""
+    device_options = config_flow.OPTIONS_SCHEMA(
+        {
+            "io": {
+                "1": "Binary Sensor",
+                "2": "Digital Sensor",
+                "3": "Switchable Output",
+            },
+            "binary_sensors": [
+                {"zone": "1", "type": "window", "name": "winder", "inverse": True},
+            ],
+            "sensors": [{"zone": "2", "type": "ds18b20", "name": "temper"}],
+            "switches": [
+                {
+                    "zone": "3",
+                    "name": "switcher",
+                    "activation": "low",
+                    "momentary": 50,
+                    "pause": 100,
+                    "repeat": 4,
+                },
+            ],
+        }
+    )
+
+    device_config = config_flow.CONFIG_ENTRY_SCHEMA(
+        {
+            "host": "1.2.3.4",
+            "port": 1234,
+            "id": "112233445566",
+            "model": "Konnected Pro",
+            "access_token": "11223344556677889900",
+            "default_options": device_options,
+        }
+    )
+
+    entry = MockConfigEntry(
+        domain="konnected", data=device_config, unique_id="112233445566"
+    )
+    entry.add_to_hass(hass)
+
+    result = await hass.config_entries.options.async_init(
+        entry.entry_id, context={"source": "test"}
+    )
+    assert result["type"] == "form"
+    assert result["step_id"] == "options_io"
+
+    # confirm the defaults are set based on current config - we"ll spot check this throughout
+    schema = voluptuous_serialize.convert(result["data_schema"])
```

Comment:
It should be enough to pass en empty dict to the schema and it should return a dict with defaults set.

Reply:
very cool - I didn't realize that field was the validation function.  Will fix

Reply:
There are some more occurrences where we can simplify this.

Reply:
got them